### ISSUE TITLE
Add a .testgroundignore kind-of feature

### ISSUE THREAD

**Issue by hacdias** [testground/testground#1170]
## Description

We should add a `.testgroundignore` kind of feature (similar to `.gitignore`). I came across this issue while building the `docker/node` builder (#1158). For Node.js projects, the `node_modules` directory is usually quite big, and uploading it to the daemon every time we need to run a test doesn't seem to be the wisest option.

Thus, I suggest creating an ignore list, in the format of a `.testgroundignore` file.

## What defines this endeavour to be complete?

- [ ] Support `.testgroundignore`

**Comment by hacdias** [comment]
@nonsense what do you think?

**Comment by nonsense** [comment]
@hacdias I think we need this.

**Comment by ackintosh** [comment]
Hi, I'm also facing this issue. I can't wait https://github.com/testground/testground/pull/1257 is released!

**Comment by laurentsenta** [comment]
+1 rust generates huge `./target` folders which make the developer experience painful

**Comment by GlenDC** [comment]
Why is a `.dockerignore` file not enough, given these builders make use of docker to build the runners? Might be a bit of a silly question, but it surprises me none the less.

**Comment by laurentsenta** [comment]
That's a good question, they solve similar problems thing but at different steps of a test lifecycle.

When we `testground run|build`:

1 - The client sends the test plan to the testground server. `.testgroundignore` is used.
2 - The server builds one or more test artifacts, sometimes with docker. `.dockerignore` might be used at the end of this step.

A user might `.dockerignore` files that Testground needs to build the test, for example:

- [custom `go.vxxx.sum` files](https://github.com/libp2p/test-plans/tree/master/ping/go) used by the `docker:go` builder,
- The [Dockerfile themselves](https://github.com/libp2p/test-plans/blob/master/ping/rust/Dockerfile)



In other words, `.testgroundignore` helps you define your test's "canonical content" which might contain assets & files that will be ignored by a `.dockerfile`.
